### PR TITLE
📚 Doc: added guide how to build docker image with fiber

### PR DESCRIPTION
**Pull request by etroynov**
## Description

I am a Golang programming mentor. In my lessons I use your wonderful framework and often my students wonder how to dockerize the end result. They are trying to find this information on Google because it is not on the official website. For this reason I created this tutorial, I hope you will like it.
## Type of change

- [x] This change requires a documentation update

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation - /docs/ directory for https://docs.gofiber.io/
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] If new dependencies exist, I have checked that they are really necessary and agreed with the maintainers/community (we want to have as few dependencies as possible)
- [x] I tried to make my code as fast as possible with as few allocations as possible
- [x] For new code I have written benchmarks so that they can be analyzed and improved

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: [CONTRIBUTING.md](https://github.com/gofiber/fiber/blob/master/.github/CONTRIBUTING.md#pull-requests-or-commits)

### CODE REVIEW

File: docs/guide/docker.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+id: docker
+title: 🚢 Build docker image
+description: Build simple docker image that you can use on the production
```
Comment (gaby) [inline docs/guide/docker.md:4]: Typo "on production"

File: docs/guide/docker.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+```bash title="~/example/Dockerfile"
+FROM golang:1.21-alpine as base
+
+WORKDIR /app
+COPY . .
+
+RUN go mod download && go mod verify
+RUN GOOS=linux GOARCH=amd64 go build -o /server .
+
+FROM gcr.io/distroless/static-debian12
```
Comment (gaby) [inline docs/guide/docker.md:74]: You are using Alpine for the based which uses `musl` and debian for the final stage which uses `glibc`. This can causr major issues depending on your application. The way to do this is:

```Dockerfile
FROM golang:1.21 as build

WORKDIR /go/src/app
COPY . .

RUN go mod download
RUN go vet -v
RUN go test -v

RUN CGO_ENABLED=0 go build -o /go/bin/app

FROM gcr.io/distroless/static-debian11

COPY --from=build /go/bin/app /
CMD ["/app"]
```

File: docs/guide/docker.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import "github.com/gofiber/fiber/v2"
+
+func main() {
+	app := fiber.New()
+
+	app.Get("/", func(c *fiber.Ctx) error {
+		return c.SendString("Hello, World!")
+	})
+
+	app.Listen(":4000")
```
Comment (gaby) [inline docs/guide/docker.md:51]: Update this to `:3000` which the typical one used for expressjs apps

File: docs/guide/docker.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+EXPOSE 4000
+CMD ["./server"]
+```
+
+:::info
+"Distroless" images contain only your application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution.
+:::
+
+:::tip
```
Comment (gaby) [inline docs/guide/docker.md:87]: This whole section is irrelevant to GoFiber. This is more for Docker docs

File: docs/guide/docker.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
+---
+id: docker
+title: 🚢 Build docker image
```
Comment (gaby) [inline docs/guide/docker.md:3]: The emoji should be a whale

**Review by gaby — CHANGES_REQUESTED** [review]

File: docs/guide/docker.md (hunk 3 above)
Comment (etroynov) [reply docs/guide/docker.md:51]: Yes, but some of frontent frameworks ( as example react ) also using this port, thats why I usually use `4000`

Comment (etroynov) [reply docs/guide/docker.md:51]: but I will change it to `3000`

File: docs/guide/docker.md (hunk 2 above)
Comment (etroynov) [reply docs/guide/docker.md:74]: fixed

File: docs/guide/docker.md (hunk 1 above)
Comment (etroynov) [reply docs/guide/docker.md:4]: fixed

File: docs/guide/docker.md (hunk 4 above)
Comment (etroynov) [reply docs/guide/docker.md:87]: Yes, that's right, but students often ask this question, why this particular image, and not, say, Docker or Ubuntu. Therefore, I decided to add this as an information section to immediately avoid possible questions in issue section.

But if it critical I can remove it.

File: docs/guide/docker.md (hunk 5 above)
Comment (etroynov) [reply docs/guide/docker.md]: fixed

File: docs/guide/docker.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
+---
+id: docker
+title: 🐳 Build docker image
+description: Build a simple docker image that can be used in production.
+sidebar_position: 8
+---
+
+After we have completed the development of our application, the moment comes when we want to upload it to the real world, most often we use Docker for this, wrapping our application in a container.
```
Comment (gaby) [inline docs/guide/docker.md:8]: Packaging instead of Wrapping. This paragraph could use some work.

File: docs/guide/docker.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+Add **Fiber** to our project:
+```bash title="~/example"
+go get github.com/gofiber/fiber/v2
+```
+
+## Step 2 - Hello world
+
+To start we shold create `main.go` file:
```
Comment (gaby) [inline docs/guide/docker.md:32]: Typo: should

File: docs/guide/docker.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+```bash title="~/example"
+touch main.go
+```
+
+Next we should add following content to our file:
```
Comment (gaby) [inline docs/guide/docker.md:37]: Syntax "the following content".

File: docs/guide/docker.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+FROM gcr.io/distroless/static-debian12
+COPY --from=build /server .
+
+ENV SOME_VAR=foo
+
+EXPOSE 3000
+CMD ["./server"]
+```
+
+:::info
```
Comment (gaby) [inline docs/guide/docker.md:83]: I'd say remove everything below this line, except the last step about running it.

File: docs/guide/docker.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+## Step 4 - build an image
+
+For this you should execute fllowing command in console:
+```bash title"~/example"
+docker build -t <your_docker_image_tag> .
+```
+
+## Step 5 - profit
+
+Now you can run you docker image by the command:
```
Comment (gaby) [inline docs/guide/docker.md:106]: Syntax "by running the following command"

**Review by gaby — CHANGES_REQUESTED** [review]
Added more comments

**Comment by ReneWerner87** [comment]
@etroynov could you solve the review comments

File: docs/guide/docker.md (hunk 6 above)
Comment (etroynov) [reply docs/guide/docker.md:8]: Sry, english is not my native language and I'm not sure what I get an idea. Can you give me an example?

File: docs/guide/docker.md (hunk 7 above)
Comment (etroynov) [reply docs/guide/docker.md:32]: fixed.

File: docs/guide/docker.md (hunk 8 above)
Comment (etroynov) [reply docs/guide/docker.md:37]: fixed.

File: docs/guide/docker.md (hunk 9 above)
Comment (etroynov) [reply docs/guide/docker.md:83]: I agree with you if we are talking about experienced developers (3-5 years of experience) but I tried to cover as large a group of developers as possible.

For example, I wouldn’t read such a section at all because I have a lot of experience with compiled languages and Docker and I can write a config using Docker documentation, but many people who studied with me usually only knew the general principles and how to run containers.

They did not know how to assemble containers and what two-stage assembly is, etc. By adding this information here, I want to close all possible questions. Because otherwise they will ask them here in "Discuss" or in stackoverflow.

**Comment by etroynov** [comment]
@ReneWerner87 sure!

File: docs/guide/docker.md (hunk 10 above)
Comment (etroynov) [reply docs/guide/docker.md]: fixed

File: docs/guide/docker.md (hunk 6 above)
Comment (gaby) [reply docs/guide/docker.md:8]: The paragraph is more suited for a blog or medium, not the documentation of a library

**Comment by etroynov** [comment]
Unfortunately, you and I have a radically opposite vision of how this article should look, because my experience suggests that such a strong reduction will lead to understatement and, as a result, additional questions, and in some moments I don’t understand what you mean.

Perhaps someone in the future will be able to complete this article, but I’m closing it. Since the end result is clearly not what I had in mind initially and I would not want people to read this article and not find additional information and think that the instructions were made in a hurry.

p.s. Just for info look how laravel with laracasts grow their community.

**Comment by gaby** [comment]
@etroynov All I asked is for a link to Docker's official documentation for anything Docker related. The official documentation explains really well what/how/why multistage builds are done. If we start writing our own thing it will get outdated and lead to people opening tickets because what we wrote in the guide is no longer valid.

**Comment by etroynov** [comment]
@gaby 

As far as I know, the Docker documentation does not contain recipes for specific technologies, but only general information about dockerization and a couple of basic examples.

Generally, people learning development (and this includes me) just want simple dockerization instructions so they can get results here and now and continue learning. How many people do you think will graduate if, after they create a simple CRUD application, I tell them: "Okay, now that you have created a simple application, you need to wrap it in a container, but to do this you need to read all the docker information, it will only take about 5-8 hours.”

Statistics show that in such cases, from 50% to 80% of students quit their studies because it is difficult.

Therefore, this article is written in a way that will suit both beginners and professionals (although I'm sure 90% of them won't need it).

The option you propose is designed for mid- and high-level developers and discards those who have just arrived, although according to statistics they are the ones who most often read the documentation.

For example, let's say you use Documentosaurus to create documentation. Please note that they use the eli5 approach to explain their technology, this is done specifically with the expectation of reaching the entire possible audience and not just pros.

As for your concern about documentation becoming outdated, this is a normal phenomenon and there is also a date for this and it is located in the guides section, which indicates that this is not part of the API but just some kind of instruction. Well, there is always us, the community, yes it will become outdated, someone will see it and create a ticket, another person will come and correct it (maybe it will be me or someone like me) this is exactly how the open source community works.

**Comment by etroynov** [comment]
@gaby 

Special respect to you for the speed of your response, this is a rarity in the open source community and it is worthy of respect.
